sql_grader/problem.py

```diff
@@ -93,25 +111,41 @@ def clone_database(source):
         return destination
 
     @classmethod
-    def run_query(cls, source, query, verify_query=None):
+    def run_query(cls, source, query, verify_query=None,
+                  modification_query=None):
         """
-        Execute the provided SQL query against a copy of the database
+        Execute the provided answer and verification queries against a copy of
+        the database
         """
-        def run(database, query):
+        def run(database, query, is_single_query):
             result = []
             message = None
             with database as connection:
                 try:
-                    for row in connection.execute(query):
+                    if is_single_query:
+                        executor_func = connection.execute
+                    else:
+                        executor_func = connection.executescript
+                    rows = executor_func(query)
```

Comment:
Didn't executescript return no rows? Then this `rows = executor_func(query)` could be used just inside the `if is_single_query`.
But then we'd have the query-running in two places of the `if` (one with storing results, another withou) and this code would get complex.

Maybe add a comment saying that rows will be returned just by `execute`, not by `executescript`? (If you're sure that it's like that). It came as a late discovery to us, so it would be useful to add a comment.